internal/infrastructure/storage/db/badger/trade_repository_impl.go

```diff
@@ -210,34 +210,45 @@ func (t tradeRepositoryImpl) getTrade(
 	ctx context.Context,
 	ID uuid.UUID,
 ) (*domain.Trade, error) {
-	var trade domain.Trade
+	var trades []domain.Trade
 	var err error
+	query := badgerhold.Where("ID").Eq(ID)
 	if ctx.Value("tx") != nil {
 		tx := ctx.Value("tx").(*badger.Txn)
-		err = t.store.TxGet(tx, ID, &trade)
+		err = t.store.TxFind(tx, &trades, query)
 	} else {
-		err = t.store.Get(ID, &trade)
+		err = t.store.Find(&trades, query)
 	}
 	if err != nil {
-		if err == badgerhold.ErrNotFound {
-			return nil, nil
-		}
 		return nil, err
 	}
 
-	return &trade, nil
+	if len(trades) <= 0 {
+		return nil, nil
+	}
+
+	return &trades[0], nil
 }
 
 func (t tradeRepositoryImpl) updateTrade(
 	ctx context.Context,
 	ID uuid.UUID,
 	trade domain.Trade,
 ) error {
+	query := badgerhold.Where("ID").Eq(ID)
 	if ctx.Value("tx") != nil {
 		tx := ctx.Value("tx").(*badger.Txn)
-		return t.store.TxUpdate(tx, ID, trade)
+		return t.store.TxUpdateMatching(tx, &domain.Trade{}, query, func(record interface{}) error {
+			tr := record.(*domain.Trade)
+			*tr = trade
+			return nil
+		})
 	}
-	return t.store.Update(ID, trade)
+	return t.store.UpdateMatching(&domain.Trade{}, query, func(record interface{}) error {
```

Comment:
This also looks like potential performance setback.(again this difference between searching row by key or query)

Reply:
This will be fixed in the PR containing the wallet detachment. For now it's needed because of the unfortunate choice of using the entire struct uuid.UUID as key for a stored trade.
Without this, it might happen that the daemon duplicates the same trade which is a serious bug that only for (un)luck has never been caught.

Reply:
Add a comment about this in the code please